evennia/contrib/aisystem/README.md

```diff
@@ -0,0 +1,661 @@
+
+`bb['tree']` - the agent's origin tree
+
+`bb['nodes'][some_node.hash]` - the dictionary of node-specific data for
+                                `some_node`
+
+`bb['globals']` - the dictionary of the agent's global data
+
+For more information on the entries of the latter two dictionaries, see the
+"Blackboards" section.
+
+Nodes automatically handle exceptions encountered within the `update` method by
+storing the error message and returning the ERROR status. Ultimately, if the
+nodes you have coded are set up properly (i.e. they return ERROR whenever one of
+their children returns ERROR) the entire tree will return the ERROR status, so
+you can check for that status within your scripts or tickerhandlers and display
+that error message accordingly. Errors are stored in the dictionary
+`bb['globals']['errors']`, whose keys are the full hash values of the nodes
+that have encountered errors. To add an error string to the dictionary, do this:
+
+```
+bb['globals']['errors'][node.hash] = error_string
+```
+
+The tree does not clean up its error strings automatically. To do so, you should
+run the agent's `ai.clean()` method:
+
+```
+agent.ai.clean()
+```
+
+Finally, you should never need to overload the node's \_\_init\_\_ method.
+
+### Limitations
+
+The theoretical upper limit for the number of nodes in a tree is 238,328.
+There is a risk of nodes failing to be generated as you approach that limit. 
+You should probably split your tree up into multiple trees connected by 
+Transition nodes well before this starts to happen, however.
+
+In theory, in extremely rare cases, a node might fail to generate a hash due to
+too many failed attempts at creating a hash that does not match the hashes of
+any nodes that already exist in the tree. Should you be informed that this
+has happened, simply create the node again.
+
+No tree, node or blackboard may be named 'this', as the name 'this' is a 
+keyword in some commands signifying the currently browsed tree or blackboard.
+Also, the names of trees, nodes and blackboards must not contain single quotes
+or equals signs, as these are also used by various commands.
+
+The names of your trees, blackboards AI agents and AI scripts should ideally
+not be numbers, as you will otherwise have to refer to them exclusively via
+their database ids in the aisystem commands.
+
+The data properties of any nodes you create should not have a __call__ method.
+If they do, they will not be displayed appropriately by the @ailook command, 
+which attempts to avoid displaying the node's methods by ignoring properties
+with a __call__ method and will also therefore fail to display your own.
+
+If there is a feature you would like added to this system, or if you find a
+bug or typo that needs fixing, don't hesitate to contact me on my github
+account, andrei-pmbcn, or at my email address, andreipmbcn at yahoo dot com.
+I am Andrei Pambuccian, the maintainer of this contrib.
```

Comment:
I think it would be excellent with a full example of a small behavior tree as a final section. 

1. Defining a small tree, only 2-3 nodes small
2. Show how to add to an AIObject in code. 
3. Example of usage in code
4. Example of usage from in-game.